<launch>

    <!--Parsing Argument-->

    <!--Setpoints-->
    <arg name="xd_StP" default="3.0" />
    <arg name="yd_StP" default="-2.0" />
    <arg name="zd_StP" default="-0.4" />

    <arg name="max_v" default="1.0" />
    <arg name="pitch_bias" default="0.000" />
    <arg name="roll_bias" default="-0.000" />

    <!--Running nodes-->
	<node name="distance_collection" pkg="coop_con_loc" type="distance_collection.py" output="screen" />	
	<node name="localization_main" pkg="coop_con_loc" type="localization_main" output="screen" />
	<node name="absolute_velocity_main" pkg="coop_con_loc" type="absolute_velocity_main" output="screen" />

	<node name="controller" pkg="coop_con_loc" type="controller_lqr.py" output="screen" >
        
        <!--Setpoints-->
        <param name="xd_StP" value="$(arg xd_StP)" />
        <param name="yd_StP" value="$(arg yd_StP)" />
        <param name="zd_StP" value="$(arg zd_StP)" />

        <param name="max_v" value="$(arg max_v)" />
        <param name="pitch_bias" value="$(arg pitch_bias)" />
        <param name="roll_bias" value="$(arg roll_bias)" />
    </node>

	<node name="px4_communication" pkg="coop_con_loc" type="px4_communication.py" output="screen" />
</launch>
